src/main/java/org/prebid/server/auction/ApPostProcessor.java

```diff
@@ -0,0 +1,77 @@
+package org.prebid.server.auction;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import org.prebid.server.cookie.UidsCookie;
+import org.prebid.server.model.HttpRequestContext;
+import org.prebid.server.proto.response.AmpResponse;
+import org.prebid.server.settings.model.Account;
+
+import javax.management.ObjectName;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class ApPostProcessor implements BidResponsePostProcessor {
+
+    public ApPostProcessor()
+    {
+    System.out.println("in constructor");
+    }
+
+    @Override
+    public Future<BidResponse> postProcess(HttpRequestContext httpRequest, UidsCookie uidsCookie,
+                                           BidRequest bidRequest, BidResponse bidResponse, Account account) {
+
+        List<SeatBid> seatBids = bidResponse.getSeatbid();
+        if(!seatBids.isEmpty()) {
+            seatBids.forEach(seatBid -> {
+                try {
+                    modifyBidTargeting(seatBid);
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
```

Comment:
This shouldnt throw. It should always return Future and never throw an exception

https://github.com/adpushup/prebid-server-java/blob/master/docs/auction-result-post-processing.md